.github/workflows/deploy.yaml

```
@@ -30,15 +30,84 @@ jobs:
+          echo "Fetching resources for stack: ${{ env.StackName }}"
+          aws cloudformation describe-stack-resources --stack-name ${{ env.StackName }} \
+            --query "StackResources[*].LogicalResourceId" --output json > resources.json
+          cat resources.json
+        else
+          echo "No resources to track since stack does not exist."
+        fi
+
+    - name: Prompt for resource deletion
+      if: ${{ github.head_ref == 'DELETESTACK' && env.STACK_NOT_FOUND == 'false' }}
+      id: resource_deletion
       run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2002:style:2:17: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead

(shellcheck)